### PR TITLE
add v510 benchmark doc

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add v510 benchmark doc

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6499
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
+---
+title: TiDB Sysbench Performance Test Report -- v5.1.0 vs. v5.0.2
+---
+
+# TiDB Sysbench Performance Test Report -- v5.1.0 vs. v5.0.2
+
+## Test overview
+
+This test aims at comparing the Sysbench performance of TiDB v5.1.0 and TiDB v5.0.2 in the Online Transactional Processing (OLTP) scenario. The results show that compared with v5.0.2, the Point Select performance of 5.1.0 is improved by 19.4%, and the performance of the Read Write and Update Index is slightly reduced.
```
Comment (TomShawn) [inline benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md:9]:
```suggestion
This test aims at comparing the Sysbench performance of TiDB v5.1.0 and TiDB v5.0.2 in the Online Transactional Processing (OLTP) scenario. The results show that compared with v5.0.2, the Point Select performance of v5.1.0 is improved by 19.4%, and the performance of the Read Write and Update Index is slightly reduced.
```

File: benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+set global tidb_enable_clustered_index = 1; 
+```
+
+## Test plan
+
+1. Deploy TiDB v5.1.0 and v5.0.2 using TiUP.
+2. Use Sysbench to import 16 tables, each table with 10 million rows of data.
+3. Execute the `analyze table` statement on each table.
+4. Back up the data used for restore before different concurrency tests, which ensures data consistency for each test.
+5. Start the Sysbench client to perform the `point_select`, `read_write`, `update_index`, and `update_non_index` tests. Perform stress tests on TiDB via AWS NLB. In each type of test, the warm-up takes 1 minute and the test takes 5 minutes.
```
Comment (TomShawn) [inline benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md:84]:
```suggestion
5. Start the Sysbench client to perform the `point_select`, `read_write`, `update_index`, and `update_non_index` tests. Perform stress tests on TiDB via HAProxy. The test takes 5 minutes.
```

File: benchmark/v5.1-performance-benchmarking-with-tpcc.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+---
+title: TiDB TPC-C Performance Test Report -- v5.1.0 vs. v5.0.2
+---
+
+# TiDB TPC-C Performance Test Report -- v5.1.0 vs. v5.0.2
+
+## Test overview
+
+This test aims to compare the TPC-H performance of TiDB 5.1.0 and TiDB 5.0.2 in the online analytical processing (OLAP) scenario. The results show that compared to v5.0.2, the TPC-C performance of v5.1.0 is improved by 2.8%.
```
Comment (TomShawn) [inline benchmark/v5.1-performance-benchmarking-with-tpcc.md:9]:
```suggestion
This test aims to compare the TPC-H performance of TiDB v5.1.0 and TiDB v5.0.2 in the online analytical processing (OLAP) scenario. The results show that compared with v5.0.2, the TPC-C performance of v5.1.0 is improved by 2.8%.
```

File: benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+| Service type   | Software version    |
+|:----------|:-----------|
+| PD        | v5.0.2 and v5.1.0   |
+| TiDB      | v5.0.2 and v5.1.0   |
+| TiKV      | v5.0.2 and v5.1.0   |
+| Sysbench  | 1.0.20     |
+
+### Parameter configuration
+
+TiDB v5.1.0 and v5.0.2 use the same configuration.
```
Comment (TomShawn) [inline benchmark/benchmark-sysbench-v5.1.0-vs-v5.0.2.md:33]:
```suggestion
TiDB v5.1.0 and TiDB v5.0.2 use the same configuration.
```

File: benchmark/v5.1-performance-benchmarking-with-tpcc.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+set global tidb_enable_async_commit = 1;
+set global tidb_enable_1pc = 1;
+set global tidb_guarantee_linearizability = 0;
+set global tidb_enable_clustered_index = 1;
+```
+
+## Test plan
+
+1. Deploy TiDB v5.1.0 and v5.0.2 using TiUP.
+2. Create a database tpcc: `create database tpcc;`.
```
Comment (TomShawn) [inline benchmark/v5.1-performance-benchmarking-with-tpcc.md:83]:
```suggestion
2. Create a database named `tpcc`: `create database tpcc;`.
```

File: benchmark/v5.1-performance-benchmarking-with-tpcc.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+3. Use BenchmarkSQL to import the TPC-C 5000 Warehouse data: `tiup bench tpcc prepare --warehouse 5000 --db tpcc -H 127.0.0.1 -p 4000`.
+4. Execute the `tiup bench tpcc run -U root --db tpcc --host 127.0.0.1 --port 4000 --time 300s --warehouses 5000 --threads {{thread}}` command to pressurize TiDB via HAProxy.
+5. Extract the tpmC data of New Order from the result.
+
+## Test result
+
+Compared to v5.0.2, the TPC-C performance of v5.1.0 is **improved by 2.8%**.
```
Comment (TomShawn) [inline benchmark/v5.1-performance-benchmarking-with-tpcc.md:90]:
```suggestion
Compared with v5.0.2, the TPC-C performance of v5.1.0 is **improved by 2.8%**.
```

**Review by TomShawn — COMMENTED** [review]
some minor comments

**Review by TomShawn — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]

**Pull request merged**